src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,217 @@
+
+Given the dynamical mapping `F`, if the iteration `x_{n+1} = 
+F(x_n)` respects the condition `T(x_{n+1}) > Tm` we accept 
+this next point, this is the _step_ part of the method. If 
+not, the method search randomly the next point in a 
+neighborhood following a given probability distribution, this 
+is the _stagger_ part. This part sometimes fails to find a new 
+candidate and a new starting point of the trajectory is chosen 
+within the defined region. See the keyword argument 
+`stagger_mode` for the different available methods.   
+
+The method produces a pseudo-trajectory of `N` points δ-close
+to the stable manifold of the chaotic saddle. 
+
+## Keyword arguments
+* `δ = 1e-10`: it is a small number constraining the random
+  search around a particular point. The interpretation of this 
+  number will depend on the distribution chosen for the 
+  generation (see `stagger_mode`).
+
+* `Tm = 30`: The minimum number of iterations of `ds` before 
+  the trajectory escapes from the bounding box defined by
+  `isinside`. 
+
+* `max_steps = 10^5`: The search for a new candidate point may 
+  fail at some point. If the search fails after `max_steps`, 
+  a new initial point is set and the method starts from a new
+  point 
+
+* `stagger_mode = :exp`: There are several ways to produce 
+  candidate points `x` that have to fulfill the condition 
+  `T(x) > Tm`. The available methods are: 
+
+    * `:exp`: An candidate following an truncated exponential 
+      distribution in a random direction `u` around the current
+      `x` such that `x_c = x + u*r`. `r = 10^-s` with `s` taken
+      from a uniform distribution in [-15, δ]. This mode fails
+      often but still manage to provide long enough stretch of
+      trajectories. 
+
+    * `:unif`: The next candidate is `x_c = x + u*r` with `r` 
+      taken from a uniform distribution [0,δ]. 
+    * `:adaptive`: The next candidate is `x_c = x + u*r` with 
+      `r` drawn from a gaussian distribution with variance  δ.
+      The variance changes according to a free parameter `f`  
+      such that `δ = δ/f` if no candidate is found and `δ = δ*f`
+      when it succeeds.    
+* `f = 1.1`: It is the free parameter for the adaptive stagger
+  method. 
+* `δ₀ = 1.`: This is the radius for the first stagger 
+trajectory search. The algorithm looks for a point 
+sufficiently close to the saddle before switching to the 
+stagger-and-step routine. The search radius must be large 
+enough to find a suitable initial. 
+[^Sweet2001]: D. Sweet, *et al.*, Phys. Rev. Lett. **86**, pp 2261  (2001)
+[^Sala2016]: M. Sala, *et al.*, Chaos **26**, pp 123124 (2016)
+"""
+function stagger_and_step!(ds::DynamicalSystem, x0, N::Int, isinside::Function; δ = 1e-10, Tm  = 30, 
+    f = 1.1, max_steps = Int(1e5), stagger_mode = :exp, δ₀ = 1.)
+
+    xi = stagger_trajectory!(ds, x0, Tm, isinside; δ₀, stagger_mode = :unif, max_steps) 
+    v = Vector{Vector{Float64}}(undef,N)
+    v[1] = xi
+@showprogress   for n in 1:N
```

Comment:
Can you change the `showprogress` to use a `ProgressMeter` and a keyword `show_progress` like we do in the rest of the library?